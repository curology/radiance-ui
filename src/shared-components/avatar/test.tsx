import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { Avatar } from '.';

describe('<Avatar />', () => {
  const imageExample =
    'data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs=';

  describe('UI snapshot', () => {
    it('renders the Avatar with default props', () => {
      const { container } = render(
        <Avatar
          src={imageExample}
          alt="avatar"
        />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('render with size medium styles', () => {
      const { container } = render(
        <Avatar
          size="medium"
          src={imageExample}
          alt="avatar"
        />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('render with size large styles', () => {
      const { container } = render(
        <Avatar
          size="large"
          src={imageExample}
          alt="avatar"
        />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });
});
