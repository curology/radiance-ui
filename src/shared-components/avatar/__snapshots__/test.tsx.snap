// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> UI snapshot render with size large styles 1`] = `
.emotion-0 {
  height: 80px;
  width: 80px;
  background-color: #B8BCFF;
  border-radius: 50%;
  overflow: hidden;
}

<img
  alt="avatar"
  class="emotion-0 emotion-1"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
/>
`;

exports[`<Avatar /> UI snapshot render with size medium styles 1`] = `
.emotion-0 {
  height: 48px;
  width: 48px;
  background-color: #B8BCFF;
  border-radius: 50%;
  overflow: hidden;
}

<img
  alt="avatar"
  class="emotion-0 emotion-1"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
/>
`;

exports[`<Avatar /> UI snapshot renders the Avatar with default props 1`] = `
.emotion-0 {
  height: 32px;
  width: 32px;
  background-color: #B8BCFF;
  border-radius: 50%;
  overflow: hidden;
}

<img
  alt="avatar"
  class="emotion-0 emotion-1"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
/>
`;
